### DISCUSSION TITLE
Login via Email or Phone Number

### DISCUSSION THREAD

**Discussion by Digitwellsolutions** [Laravel-Backpack/community-forum#745] [General]
Hi,
   I have extended the user table with phonenumber and phonecountry columns to allow the user to login with a phone number instead of an email.
I have made the change to the view, login.blade.php, but I can't see how to change the code to use this information.

The blade goes to route - backpack.auth.login on submit, 

according to the route:list, this goes to 
 GET|HEAD        admin/login .................................................... backpack.auth.login › Backpack\CRUD › LoginController@showLoginForm
 
but I can't find this function. 

Any help/ tips would be greatly appreciated.

**Comment by jcastroa87** [comment]
Hi @Digitwellsolutions 

This function is in /vendor/backpack/crud/src/app/Library/Auth/AuthenticatesUsers.php

You need to override if need to customize the login process.

Cheers.

**Comment by Digitwellsolutions** [comment]
Thanks @jorgetwgroup  for the steer.
I had to:
1. Create my own LoginController and also a copy of AuthenticateUsers trait which this uses.
2. Make the appropriate changes
3. set 'setup_auth_routes' to false
4. copy all of the login and register routes to 'routes/backpack/custom.php'
5. Point the login routes at my new controller.

I now need to do the same for forgotten password.

Is there an easier way?

**Reply by jcastroa87** [reply]
There is a way to use a username instead email [here](https://backpackforlaravel.com/docs/6.x/base-how-to#login-with-username-instead-of-email-1), but if you want to customize more you need to handle in this way.

Cheers.

**Comment by jcastroa87** [comment]
I will close the issue, if you need any further support with this topic, you are free to re-open or create a new one.

Cheers.
